### PR TITLE
cargo clippy --fix

### PR DESCRIPTION
**Pull request by sporksmith**
`cargo clippy --fix` can be helpful for fixing clippy warnings. There are currently a lot of things in our code it changes though, making it awkward to use in a PR since it'll change code unrelated to the PR.

I'm not sure why, but `cargo clippy --fix` warns about and fixes things that just `cargo clippy` doesn't complain about. Maybe it has a different default policy set.

This PR includes some manual fixes as well for things that `clippy --fix` fixes suboptimally, or complains about without fixing.

### CODE REVIEW

File: src/lib/shim/src/tls.rs [hunk 1]
```diff
@@ -742,12 +742,12 @@ mod test {
...
 
                         // Order shouldn't matter here, but change it from above anyway.
-                        my_i32.get().store(i as i32, atomic::Ordering::Relaxed);
+                        my_i32.get().store(i, atomic::Ordering::Relaxed);
                         my_i8.get().store(i as i8, atomic::Ordering::Relaxed);
                         my_i16.get().store(i as i16, atomic::Ordering::Relaxed);
 
                         assert_eq!(my_i16.get().load(atomic::Ordering::Relaxed), i as i16);
-                        assert_eq!(my_i32.get().load(atomic::Ordering::Relaxed), i as i32);
+                        assert_eq!(my_i32.get().load(atomic::Ordering::Relaxed), { i });
```
Comment (stevenengler) [inline src/lib/shim/src/tls.rs:750]: why the braces?

File: src/lib/vasi-sync/tests/scchannel-tests.rs [hunk 2]
```diff
@@ -31,9 +31,7 @@ mod scchannel_tests {
                     channel.send(42);
                 })
             };
-            let reader = {
-                sync::thread::spawn(move || channel.receive())
-            };
+            let reader = { sync::thread::spawn(move || channel.receive()) };
```
Comment (stevenengler) [inline src/lib/vasi-sync/tests/scchannel-tests.rs]: I think we can remove the braces?

File: src/lib/vasi-sync/tests/scchannel-tests.rs [hunk 3]
```diff
@@ -98,9 +96,7 @@ mod scchannel_tests {
             };
             // Simulate a separate watchdog thread that detects that the writer process
             // has exited, and closes the channel in parallel with the other operations.
-            let watchdog = {
-                sync::thread::spawn(move || channel.close_writer())
-            };
+            let watchdog = { sync::thread::spawn(move || channel.close_writer()) };
```
Comment (stevenengler) [inline src/lib/vasi-sync/tests/scchannel-tests.rs]: I think we can remove the braces?

File: src/main/network/router/codel_queue.rs [hunk 4]
```diff
@@ -406,32 +406,29 @@ mod tests {
         // Not above target so interval is not set.
         assert!(cdq.interval_end.is_none());
         let now = start + TARGET - one;
-        assert_eq!(cdq.process_standing_delay(&now, TARGET - one), false);
+        assert!(!cdq.process_standing_delay(&now, TARGET - one));
         assert!(cdq.interval_end.is_none());
```
Comment (stevenengler) [inline src/main/network/router/codel_queue.rs:410]: iirc I think this used the true/false for readability. (It's easier to see the difference between `assert_eq(long_string, true)` and `assert_eq(long_string, false)` compared to `assert(long_string)` and `assert(!long_string)`.) My opinion leans slightly towards the true/false, but it doesn't really matter to me.

**Review by stevenengler — APPROVED** [review]

File: src/main/network/router/codel_queue.rs (hunk 4 above)
Comment (sporksmith) [reply src/main/network/router/codel_queue.rs:410]: Ah right. I don't have a preference of one vs the other, so lean towards it not being worth fighting clippy/convention here

File: src/lib/shim/src/tls.rs (hunk 1 above)
Comment (sporksmith) [reply src/lib/shim/src/tls.rs]: Removed

File: src/lib/vasi-sync/tests/scchannel-tests.rs (hunk 2 above)
Comment (sporksmith) [reply src/lib/vasi-sync/tests/scchannel-tests.rs]: Removed

File: src/lib/vasi-sync/tests/scchannel-tests.rs (hunk 3 above)
Comment (sporksmith) [reply src/lib/vasi-sync/tests/scchannel-tests.rs]: Removed

**Pull request merged**